lua/hbac/telescope/actions.lua

```diff
@@ -21,45 +17,31 @@ local function execute_telescope_action(prompt_bufnr, action)
 		action(single_selection.value)
 	end
 
-	local row = picker:get_selection_row()
-	picker:register_completion_callback(function()
-		picker:set_selection(row)
-	end)
 	picker:refresh(finder(finder_opts), { reset_prompt = false })
 end
 
-local function hbac_toggle_selections(prompt_bufnr)
-	execute_telescope_action(prompt_bufnr, state.toggle_pin)
+local function hbac_toggle_pin(prompt_bufnr)
+	execute_telescope_action(prompt_bufnr, hbac_actions.toggle_pin)
 end
 local function hbac_pin_all(prompt_bufnr)
-	execute_telescope_action(prompt_bufnr, actions.pin_all)
+	execute_telescope_action(prompt_bufnr, hbac_actions.pin_all)
 end
 local function hbac_unpin_all(prompt_bufnr)
-	execute_telescope_action(prompt_bufnr, actions.unpin_all)
+	execute_telescope_action(prompt_bufnr, hbac_actions.unpin_all)
 end
 local function hbac_close_unpinned(prompt_bufnr)
-	execute_telescope_action(prompt_bufnr, actions.close_unpinned)
+	execute_telescope_action(prompt_bufnr, hbac_actions.close_unpinned)
 end
 local function hbac_delete_buffer(prompt_bufnr)
-	execute_telescope_action(prompt_bufnr, config.values.close_command)
+	local close_command = require("hbac.config").values.close_command
+	execute_telescope_action(prompt_bufnr, close_command)
 end
 
-M.attach_mappings = function(_, map)
-	local hbac_telescope_actions = {
+
+return {
```

Comment:
Is it important to not use M here? or why did you change it?

Reply:
No we can still use it! Just oversimplifying I guess, or I was trying to imitate hbac.init.lua ? The important part is defining the named local functions and assigning them as values to the keys in the table (unless there's a better way). Just as long as we're not doing e.g.

```lua
M.pin_all = function(prompt_bufnr) -- etc. 
end

return M
```

Will change it back